src/jit/MemoryInl.h

```diff
@@ -1390,15 +1391,70 @@ static void emitAtomic(sljit_compiler* compiler, Instruction* instr)
         sljit_s32 baseReg = SLJIT_EXTRACT_REG(addr.memArg.arg);
         sljit_s32 tmpReg = srcReg;
 
+        JITArg memValue(operands + 0);
+        sljit_s32 memValueReg = SLJIT_EXTRACT_REG(memValue.arg);
+        sljit_s32 maskReg;
+        sljit_s32 shiftReg;
+
+        if (noShortAtomic && size <= 2) {
+            operationSize = SLJIT_MOV32;
+
+            shiftReg = instr->requiredReg(2);
+            maskReg = instr->requiredReg(3);
+        }
+
+        if (SLJIT_IS_IMM(memValueReg)) {
+            return;
+        }
+
+        if (noShortAtomic && size <= 2) {
+            sljit_emit_op2(compiler, SLJIT_AND32, shiftReg, 0, baseReg, 0, SLJIT_IMM, 0x3);
```

Comment:
Isn't baseReg is a word sized reg?

Reply:
it is; would the more proper way to do this be using `SLJIT_AND` here, and then using an `SLJIT_MOV32` from and to `shiftReg`?